cmd/query/app/analytics/http_gateway.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package analytics
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel/trace"
+	"go.uber.org/zap"
+)
+
+const (
+	routeGetDeepDependencies = "/api/deep-dependencies"
+)
+
+// HTTPGateway exposes analytics HTTP endpoints.
+type HTTPGateway struct {
+	Logger *zap.Logger
+}
+
+func (h *HTTPGateway) RegisterRoutes(router *mux.Router) {
+	h.addRoute(router, h.getDeepDependencies, routeGetDeepDependencies).Methods(http.MethodGet)
+}
+
+// addRoute adds a new endpoint to the router with given path and handler function.
+// This code is mostly copied from ../http_handler.
+func (*HTTPGateway) addRoute(
+	router *mux.Router,
+	f func(http.ResponseWriter, *http.Request),
+	route string,
+	_ ...any, /* args */
+) *mux.Route {
+	var handler http.Handler = http.HandlerFunc(f)
+	handler = otelhttp.WithRouteTag(route, handler)
+	handler = spanNameHandler(route, handler)
+	return router.HandleFunc(route, handler.ServeHTTP)
+}
+
+func spanNameHandler(spanName string, handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		span := trace.SpanFromContext(r.Context())
+		span.SetName(spanName)
+		handler.ServeHTTP(w, r)
+	})
+}
+
+func (h *HTTPGateway) getDeepDependencies(w http.ResponseWriter, _ *http.Request) {
+	dependencies := TDdgPayload{
+		Dependencies: []TDdgPayloadPath{
+			{
+				Path: []TDdgPayloadEntry{
+					{Service: "sample-serviceA", Operation: "sample-opA"},
+					{Service: "sample-serviceB", Operation: "sample-opB"},
+				},
+				Attributes: []Attribute{
+					{Key: "exemplar_trace_id", Value: "abc123"},
+				},
+			},
+		},
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(dependencies); err != nil {
```

Comment:
1. encode in memory
2. if error return 501
3. otherwise write content-type and output